Выполнены задания под ТЗ-7:

Vladyak1 opened the pull request:
- добавлен csvTableHeaders с enum значениями;
- добавлен ManagerSaveException;
- добавлена папка utils, куда был перемещён Status, изменены ссылки импорта под него;
- создан FileBackedTaskManager, который реализует функционал чтения/записи файлов, создаёт строки из задач и наоборот

Vladyak1 committed:
Выполнены задания под ТЗ-7:
- добавлен csvTableHeaders с enum значениями;
- добавлен ManagerSaveException;
- добавлена папка utils?, куда был перемещён Status, изменены ссылки импорта под него
- создан FileBackedTaskManager, который реализует функционал чтения/записи файлов, создаёт строки из задач и наоборот

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/model/Epic.java:
```diff
@@ -12,6 +12,10 @@ public Epic(String title, String description) {
     }
 
+    public String getType() {
+        return "EPIC";
```
Кажется здесь давно пора использовать enum

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    String toString(Task task) {
```
Метод должен быть публичный и должна быть необязательная аннотация Override

VladimirTitov4 replied:
И зачем вам в принципе метод toString в этом классе ? 
Метод toString должен быть определен у каждого типа задачи (мое мнение, если не согласны давайте обсудим)

Vladyak1 replied:
Необязательная аннотация Override не получается, так как в родительском классе такой функции нет.
Метод был добавлен, так как на него указывает подсказка в ТЗ (первая подсказка "Как сохранять задачи в файл и считывать их из него", пункт 2)
За этот метод не держусь, могу легко убрать

VladimirTitov4 replied:
Ок можно оставить, но тогда лучше метод переименовать в printTasks или что-то вроде того. А то метод toString это по сути зарезервированный термин из класса Object

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,200 @@
+                writer.write(task.toString() + "\n");
+            }
+        } catch (IOException e) {
+            throw new ManagerSaveException("Ошибка при сохранении данных в файл " + filePath);
```
Правильно, что выкидываете здесь ManagerSaveException

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/utils/csvTableHeaders.java:
```diff
@@ -0,0 +1,10 @@
+package ru.yandex.practicum.tasktracker.utils;
+
+public enum csvTableHeaders {
+    id,
```
Значения enum записываются в UPPER_CASE

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/service/ManagerSaveException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.yandex.practicum.tasktracker.service;
+
+public class ManagerSaveException extends RuntimeException {
+    ManagerSaveException(String message) {
```
Конструктор должен быть публичным, иначе у вас package-level access

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,200 @@
+             }
+    }
+
+    String historyToString(HistoryManager manager) {
```
Возможно вы здесь намеренно не ставите модификатор доступа, но кажется метод должен быть публичным

Vladyak1 committed:
Добавлены исправления под ТЗ-7:
- создан класс с enum, куда переместились Status и csvTableHeaders и добавлен TaskType, enum csvTableHeaders были перезаписаны в UPPER_CASE;
- выводы отлова ошибок IOException были заменены с "sout" на "throw new";
- historyToString стал public;
- добавлен класс ManagerLoadException, а конструктор ManagerSaveException стал public.

VladimirTitov4 commented on src/ru/yandex/practicum/tasktracker/utils/enums.java:
```diff
@@ -0,0 +1,24 @@
+package ru.yandex.practicum.tasktracker.utils;
+
+public class enums {
+    public enum Status {
```
Нет, так enum не объявляется. 
Вам нужен отдельный файл для кажлого типа enum

И название класса всегда с большой буквы, если уж такой класс сделали

Vladyak1 committed:
Новые исправления для ТЗ-7:
- класс enum был расформирован, каждый enum из него теперь отдельный класс
